paper-api/src/main/java/io/papermc/paper/registry/data/client/ClientAsset.java

```diff
@@ -0,0 +1,66 @@
+package io.papermc.paper.registry.data.client;
+
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.key.KeyPattern;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Contract;
+
+/**
+ * A data-centric version-specific representation of a client asset, composed of an identifier and a path.
+ * Follows the same, version-specific, compatibility guarantees as the RegistryEntry types it is used in.
+ */
+@ApiStatus.Experimental
+@ApiStatus.NonExtendable
+public interface ClientAsset {
```

Comment:
I feel like something like `TextureAsset` would be a better name, as the api already uses other client assets like item models, which aren't represented by/with this.

Reply:
Hmmm, yea *maybe*. ClientTextureAsset?
I agree it is a bit weird that some things are only represented via a Key, others by this new type, but that is kinda how internals go and the registry data API is pretty much an API mirror of those.

Reply:
Yea, I guess the method already includes "texture" in there.
If things change down the line, we can rename it anyway as this isn't stable API.